sequencer/src/api.rs

```diff
@@ -434,24 +430,54 @@ pub mod test_helpers {
         P: PersistenceOptions,
         C: StateCatchup + 'static,
     {
+        /// NOTICE: This function will reset most of the config. Please assign the number of the nodes
+        /// before assigning values.
+        pub fn num_nodes<const N: usize>(
+            self,
+        ) -> TestNetworkConfigBuilder<N, no_storage::Options, MockStateCatchup> {
+            TestNetworkConfigBuilder {
+                state: std::array::from_fn(|_| ValidatedState::default()),
+                persistence: Some([no_storage::Options; N]),
+                catchup: Some(std::array::from_fn(|_| MockStateCatchup::default())),
+                network_config: None,
+                api_config: self.api_config,
+            }
+        }
+
         pub fn states(mut self, state: [ValidatedState; NUM_NODES]) -> Self {
             self.state = state;
             self
         }
 
-        pub fn persistences(mut self, persistence: [P; NUM_NODES]) -> Self {
-            self.persistence = Some(persistence);
-            self
```

Comment:
Why did we replace this assignment with the construction of a whole new object? The new version is more verbose and will have to be modified any time the unrelated fields are changed

Reply:
Using the new version we can set the `PersistenceOptions` dynamically and it can make the construction more simple. We can `ConfigBuilder::default()` and set the default persistences and then set it with the other type. The old version requires us to assign the type and we can never change it.

Reply:
Ohh I see, clever